pdm/cli/commands/run.py

```diff
@@ -196,10 +212,46 @@ def _run_task(self, task: Task, args: Sequence[str] = ()) -> int:
             err=True,
             verbosity=termui.DETAIL,
         )
+        if kind == "composite":
+            env = options.get("env", {})
+            env_file = options["env_file"]
+            for script in value:
+                if script is None:
+                    return 1
+                code = self._run_subtask(script, env, env_file)
```

Comment:
Ideally, the composite task should accept all options and pass them to the subtasks, rather than supporting `env` and `env_file` only.

Reply:
Changed: `_run_subtask` and `run` have been merged and take a `TaskOptions` as optionnal parameter